config/v1/attribute_descriptor.proto

```diff
@@ -0,0 +1,65 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package config.v1;
+
+// A definition of an attribute type.
+//
+// An attribute carries a piece of information describing ingress or egress traffic, along
+// with the environment the traffic takes place in.
+//
+// This descriptor represents the schema for an individual attribute type.
+message AttributeDescriptor {
+  // The name of this descriptor, referenced from individual attribute instances and other
+  // descriptors.
+  //
+  // The format of this name is:
+  //
+  //     Name = IDENT { "." IDENT } ;
+  //
+  // Where `IDENT` must match the regular expression `[a-z][a-z0-9]+`.
+  string name = 1;
+
+  // The type of data carried by attributes
+  ValueType value_type = 2;
+
+  // An optional human-readable description of the attribute's purpose.
+  string description = 3;
+
+  // Types of supported attribute values.
+  enum ValueType {
+    // A 64-bit signed integer.
+    INT64 = 0;
```

Comment:
Are we trying to represent semantic types (IPv4) or data types (INT64) here?

Is the goal of this enum to represent the semantics of of values that are both lables and metrics or just labels.
We should clarify the semantic distinction between labels and metrics.

metrics -> any value where the primary intent is to be able to aggregate it or apply a numerical threshold to it
labels -> any value where the primary intent is to use it as a dimension for aggregation of a metric or as a key for a rule

Generally speaking if there is confusion about a value being a metric or a label we should err on the side of using labels even though specialized arithmetic may be done on the value to from another label dimension. Examples
- ip address ranges to determine internal vs external IP
- http status code ranges to classify error classes